packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -327,6 +328,12 @@ export default function VisualEditor( { styles } ) {
 		[ styles ]
 	);
 
+	// Add some styles for alignwide/alignfull Post Content and its children.
+	const alignCSS = `.is-root-container.alignwide { max-width: var(--wp--style--global--wide-size); margin-left: auto; margin-right: auto;}
```

Comment:
That was added to address [this comment](https://github.com/WordPress/gutenberg/pull/49565/files#r1156707168). Post Content layout in the front end is influenced by both its layout and alignment; this was an attempt make the post editor accurately reflect them.

Reply:
So are you saying that we try to apply the "alignment" of the post content block in the post editor? For me that seems a bit random because we don't really know the layout of its parent block, maybe it's within a small sidebar, and wide there means something else. Anyway, I'm not touching that at the moment but I feel like it's just complexity for nothing.

Reply:
True, we're not looking at the complete nesting tree. It would be possible to grab any custom wide align value from the parent (which we're not doing) and that might make things a bit more accurate, but in terms of effort vs effect the current solution will work for the majority of cases. Many themes nest Post Content inside one or more Group blocks; even those that nest it inside Columns will most likely have it in a responsive column that takes up most of the page. Having Post Content display in a narrow column on the single post template is a very unlikely scenario.

With these layout problems there are always many variables at play: nesting structure, parent block styles, theme styles, root padding, alignments... and often no approach works for all cases; I think if there's no universal solution it's best to try for a solution that works for _most_ cases 😅